api/v1beta2/artifact_types.go

```diff
@@ -56,15 +56,24 @@ type Artifact struct {
 	Size *int64 `json:"size,omitempty"`
 }
 
-// HasRevision returns true if the given revision matches the current Revision
-// of the Artifact.
+// HasRevision returns if the given revision matches the current Revision of
+// the Artifact.
 func (in *Artifact) HasRevision(revision string) bool {
 	if in == nil {
 		return false
 	}
 	return in.Revision == revision
 }
 
+// HasChecksum returns if the given checksum matches the current Checksum of
+// the Artifact.
+func (in *Artifact) HasChecksum(checksum string) bool {
+	if in == nil {
+		return false
+	}
+	return in.Checksum == checksum
```

Comment:
```suggestion
	return in != nil && in.Checksum == checksum
```

Reply:
The reason I do not use this for `nil` checks is that when logic changes, there is a chance someone accidentally omits the `in != nil` check, and introduces a panic.